synapse/storage/databases/main/roommember.py

```diff
@@ -220,16 +220,12 @@ def _get_subset_users_in_room_with_profiles(
             txn: LoggingTransaction,
         ) -> Dict[str, ProfileInfo]:
             clause, ids = make_in_list_sql_clause(
-                self.database_engine, "c.state_key", user_ids
+                self.database_engine, "user_id", user_ids
             )
 
             sql = """
-                SELECT state_key, display_name, avatar_url FROM room_memberships as m
-                INNER JOIN current_state_events as c
-                ON m.event_id = c.event_id
-                AND m.room_id = c.room_id
-                AND m.user_id = c.state_key
-                WHERE c.type = 'm.room.member' AND c.room_id = ? AND m.membership = ? AND %s
```

Comment:
I think this is equivalent to the old thing. Please double-check

Reply:
The problem with this is that there are multiple entries for a given `user_id` and we don't know which is the latest just from the `room_memberships` table alone

Example:
```
SELECT user_id, display_name, avatar_url FROM room_memberships
WHERE room_id = '!abc:matrix.org' AND membership = 'join';
           user_id            |         display_name          |                avatar_url
------------------------------+-------------------------------+-------------------------------------------
 @madlittlemods:matrix.org    | madlittlemods                 |
 @madlittlemods:matrix.org    | madlittlemods                 | mxc://matrix.org/bDayqThxTIcGNcskzIADknRv
 @madlittlemods:matrix.org    | madlittlemods (Eric Eastwood) | mxc://matrix.org/bDayqThxTIcGNcskzIADknRv
 ```